File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/ChargeItem.ts
import { Annotation } from "./Annotation";
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { Timing } from "./Timing";
import { Quantity } from "./Quantity";
import { DomainResource } from "./DomainResource";
import { Money } from "./Money";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type ChargeItemPerformer = BackboneElement & {
    actor: Reference;
    function?: CodeableConcept;
};

export type ChargeItem = DomainResource & {
    service?: Reference[];
    definitionUri?: string[];
    enterer?: Reference;
    requestingOrganization?: Reference;
    definitionCanonical?: string[];
    bodysite?: CodeableConcept[];
    costCenter?: Reference;
    note?: Annotation[];
    account?: Reference[];
    reason?: CodeableConcept[];
    product?: CodeableConcept | Reference;
    supportingInformation?: Reference[];
    status: string;
    code: CodeableConcept;
    identifier?: Identifier[];
    context?: Reference;
    quantity?: Quantity;
    partOf?: Reference[];
    priceOverride?: Money;
    enteredDate?: string;
    overrideReason?: string;
    performingOrganization?: Reference;
    subject: Reference;
    factorOverride?: number;
    occurrence?: Timing | Period | string;
    performer?: ChargeItemPerformer[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/ChargeItem.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import Period
from base import CodeableConcept
from base import Timing
from base import Quantity
from base import DomainResource
from base import Money
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class ChargeItemPerformer(BackboneElement):
    actor: Reference
    function: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class ChargeItem(DomainResource):
    service: Optional[List[Reference]] = field(default_factory=list)
    definition_uri: Optional[List[str]] = field(default_factory=list)
    enterer: Optional[Reference] = None
    requesting_organization: Optional[Reference] = None
    product_codeable_concept: Optional[CodeableConcept] = None
    product_reference: Optional[Reference] = None
    definition_canonical: Optional[List[str]] = field(default_factory=list)
    bodysite: Optional[List[CodeableConcept]] = field(default_factory=list)
    occurrence_timing: Optional[Timing] = None
    cost_center: Optional[Reference] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    account: Optional[List[Reference]] = field(default_factory=list)
    reason: Optional[List[CodeableConcept]] = field(default_factory=list)
    supporting_information: Optional[List[Reference]] = field(default_factory=list)
    occurrence_period: Optional[Period] = None
    status: str
    code: CodeableConcept
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    context: Optional[Reference] = None
    quantity: Optional[Quantity] = None
    part_of: Optional[List[Reference]] = field(default_factory=list)
    price_override: Optional[Money] = None
    entered_date: Optional[str] = None
    occurrence_date_time: Optional[str] = None
    override_reason: Optional[str] = None
    performing_organization: Optional[Reference] = None
    subject: Reference
    factor_override: Optional[float] = None
    performer: Optional[List[ChargeItemPerformer]] = field(default_factory=list)